mempool/src/shared_mempool/coordinator.rs

```diff
@@ -67,6 +67,17 @@ pub(crate) async fn coordinator<V>(
     let workers_available = smp.config.shared_mempool_max_concurrent_inbound_syncs;
     let bounded_executor = BoundedExecutor::new(workers_available, executor.clone());
 
+    let initial_reconfig = mempool_reconfig_events
+        .next()
+        .await
+        .expect("Reconfig sender dropped, unable to start mempool");
+    handle_mempool_reconfig_event(
```

Comment:
Is this really necessary? It'll be fetched and handled in the loop below, no?

Reply:
> This is necessary for mempool to correctly use onchain configs (which is needed for quorum store).

I assume you're saying it's not okay for it to happen in the loop because we might get a quorum store message before the code is updated?

Reply:
If we need to read from an onchain config (as we'll do for quorum store rollout) there's a race here for each time a node is restarted where despite having already read the new config, mempool could start with the old config.

Currently, all that's done on a reconfig is restart the VM. It probably doesn't hurt too much, but restarting immediately after starting the loop seems a bit weird.

Epoch Manager takes the same approach: https://github.com/aptos-labs/aptos-core/blob/main/consensus/src/epoch_manager.rs#L901

Reply:
@bchocho and I synced offline and confirmed there's no issue around the race condition here, so forcing the config change on startup isn't required (i.e., nothing will break). But, this is more of an optimization for reboots after the on-chain config has been updated 😄

Reply:
it's more to prepare for the quorum store config right?

Reply:
I updated the description to explain a bit:

> For up-to-date nodes, the initial read will prevent a race on restart where mempool could initially handle transactions with old onchain configs. (Note though, this provides no guarantees for stale nodes or nodes that are currently undergoing an onchain config change.)

When a node that already read the onchain config restarts -- without this change, it could do the behavior of the old config for some transactions until the onchain config is read in the loop. With this change, as long as the node is up-to-date we won't expect it to use the old behavior.